gems/smithy-xml/lib/smithy-xml/parser.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require_relative 'parser/frame'
+require_relative 'parser/stack'
+
+module Smithy
+  module Xml
+    # @api private
+    class Parser
+      # Raised when an XML parsing error occurs.
+      class ParseError < StandardError
+        def initialize(msg, line, column)
+          @line = line
+          @column = column
+          super(msg)
+        end
+
+        # @return [Integer, nil]
+        attr_reader :line
+
+        # @return [Integer, nil]
+        attr_reader :column
+      end
+
+      def initialize(options = {})
+        @engine = options[:engine] || self.class.engine
+      end
```

Comment:
In smithy-json, engines are defined within `Smithy::Json` namespace - any strong preference between two setups? I guess it actually makes more sense to be under Parser since it's really used for those intentions

Reply:
Yeah. I don't think we necessarily have to follow symmetry here. The engine is for parsing only, and I think we could eventually promote it to be for building too if we wanted to, but I don't think we should let a builder handle this, at least from the client pov. For servers, possibly.